src/Components/Admin/ObjectEditor/object.js

```diff
@@ -0,0 +1,263 @@
+            setPuzzleType(data.animations_json.blackboardData.componentType);
+            setIsInteractable(data.is_interactable);
+            const puzBody = data.animations_json.blackboardData;
+            if (puzBody.componentType === "text-pane") {
+                const tempTexts = puzBody.jsonData.data;
+                for (let i = 0; i < puzBody.jsonData.data.length; i++) {
+                    tempTexts[i]["index"] = i;
+                }
+                setTexts(tempTexts);
+            }
+        };
+        if (!puzzleBody.componentType) {
+            getPuzzleBody();
+        }
+    }, [sceneId, objectId, puzzleBody, handleError]);
+
+    const selectPuzzleType = (obj) => {
+        if (obj) {
+            const data = puzzleBody;
+            data.componentType = obj.value;
+            setPuzzleBody(data);
+            setPuzzleType(obj.value);
+        }
+    };
+
+    const reorder = (texts, startIndex, endIndex) => {
+        const result = texts;
+        const [removed] = result.splice(startIndex, 1);
+        result.splice(endIndex, 0, removed);
+        return result;
+    };
+
+    const reorderTransitions = (sourceIndex, destinationIndex) => {
+        if (
+            sourceIndex == null ||
+            destinationIndex == null ||
+            sourceIndex === destinationIndex
+        ) {
+            return;
+        }
+
+        const reorderedList = reorder(texts, sourceIndex, destinationIndex);
+
+        setTexts([...reorderedList]);
+    };
+
+    const addTransition = () => {
+        const newTransition = {
+            text: prompt("Enter the text for the transition: "),
+            index: texts.length,
+        };
+
+        setTexts([...texts, newTransition]);
+    };
+
+    const onMoveUpClick = (index) => {
+        reorderTransitions(index, Math.max(0, index - 1));
+    };
+
+    const onMoveDownClick = (index) => {
+        reorderTransitions(index, Math.min(texts.length - 1, index + 1));
+    };
+
+    const deleteTransition = (index) => {
+        const tempTexts = JSON.parse(JSON.stringify(texts));
+        tempTexts.splice(index, 1);
+        for (let i = 0; i < tempTexts.length; i++) {
+            tempTexts[i].index = i;
+        }
+        setTexts(tempTexts);
+    };
+
+    const handleSubmit = () => {
+        const animCopy = animationsJson;
+        const puzzleBodyCopy = puzzleBody;
+        const textsCopy = texts;
+        for (let i = 0; i < textsCopy.length; i++) {
+            delete textsCopy["index"];
+        }
+        puzzleBodyCopy.jsonData.data = textsCopy;
+        animCopy.blackboardData = puzzleBodyCopy;
+        setAnimationsJson(animCopy);
+
+        const savePuzzle = async () => {
+            await editPuzzle(
+                { sceneId, objectId, isInteractable, animationsJson },
+                handleError
+            );
+        };
+
+        savePuzzle();
+    };
+
+    return (
+        <div className={classes.container}>
+            {isInteractable ? (
```

Comment:
Missing functionality: toggling whether an object is `isInteractable` or not should also be present (it was present in the original inspector PR, I think it didn't get migrated here?)